### PR TITLE
[UI-48] Reinvent combo boxes

### PR DESCRIPTION
**Pull request by rangoo94**
#### Task

- **Issue or task referred:** UI-48
- **Feature or Bugfix:** Feature

Combo boxes reinvented, to make it simpler for use.

#### Checklist

- [x] Combo box
- [x] Select box
- [x] Multi selection
- [x] Auto complete HOC
- [x] Required unit tests prepared
- [x] Documentation prepared
- [x] External licenses validated (ideally MIT)

### CODE REVIEW

File: packages/combo-box/src/ComboBox.js [hunk 1]
```diff
@@ -1,137 +1,388 @@
...
+  options: PropTypes.array.isRequired,
+
+  /** Placeholder to show when there is no value selected */
+  placeholder: PropTypes.node,
+
+  /** Is it multi-select? */
+  multi: PropTypes.bool,
+
+  /** Event called after current value of combo-box has been changed */
   onChange: PropTypes.func,
```
Comment (maciek15) [inline packages/combo-box/src/ComboBox.js]: Missing information about onFocus and onBlur. (Same issue inside AutoComplete.js)

File: packages/combo-box/styles/geometry.sass [hunk 2]
```diff
@@ -1,88 +1,169 @@
...
+    padding: 0
+    font: inherit
 
-  &--open:not(#{&}__input--separated) input,
-  &--open:not(#{&}__button--separated)
-    border-radius: $dropdown-border-radius $dropdown-border-radius 0 0 !important
-    border-bottom-width: 0 !important
+  &__value--placeholder
+    &, input
+      color: #bbb
```
[the remark this answers is not in the document]
Comment (rangoo94) [reply packages/combo-box/styles/geometry.sass]: I don't agree. Theming file is not only for colors. As I mentioned in other PR, probably we will get rid of theming for now

File: packages/combo-box/styles/geometry.sass [hunk 3]
```diff
@@ -1,88 +1,169 @@
...
+    position: relative
+    box-sizing: border-box
     display: flex
-    flex-direction: row
-    justify-content: space-between
     align-items: center
+    text-align: inherit
+    text-align: start
+    line-height: 1.2em
+    border: 1px solid #eee
```
[the remark this answers is not in the document]
Comment (rangoo94) [reply packages/combo-box/styles/geometry.sass]: I left it this way for now, as we need to redesign all components anyway

**Comment by rangoo94** [comment]
Fixed changes from code review, so I am merging it now - we have too many things waiting for that.

**Pull request merged**